### PR TITLE
Transceiver eeprom dom CLI modification to show output from TRANSCEIVER_DOM_THRESHOLD table

### PR DESCRIPTION
**Pull request by mihirpat1**
#### What I did
This PR is needed along with https://github.com/sonic-net/sonic-platform-daemons/pull/320 to continue supporting "show interface transceiver eeprom \<port\> -d" CLI since [PR#320](https://github.com/sonic-net/sonic-platform-daemons/pull/320) is separating DOM threshold related information in a separate table entry (TRANSCEIVER_DOM_THRESHOLD from TRANSCEIVER_DOM_SENSOR) in state DB.
The change with this PR will ensure that the CLI output remains unchanged until https://github.com/sonic-net/sonic-buildimage/issues/12911 is committed which will all together publish DOM threshold related information in a new CLI.

#### How I did it
dom_info_dict is now populated using information from both TRANSCEIVER_DOM_SENSOR as well as TRANSCEIVER_DOM_THRESHOLD.

#### How to verify it
Please refer to the enclosure attached to https://github.com/sonic-net/sonic-platform-daemons/pull/320

#### New command output (if the output of a command-line utility has changed)

Signed-off-by: Mihir Patel <patelmi@microsoft.com>

### CODE REVIEW

File: scripts/sfpshow [hunk 1]
```diff
@@ -391,6 +391,7 @@ class SFPShow(object):
                 if dump_dom:
                     sfp_type = sfp_info_dict['type']
                     dom_info_dict = state_db.get_all(state_db.STATE_DB, 'TRANSCEIVER_DOM_SENSOR|{}'.format(interface_name))
+                    dom_info_dict.update(state_db.get_all(state_db.STATE_DB, 'TRANSCEIVER_DOM_THRESHOLD|{}'.format(interface_name)))
```
Comment (shyam77git) [inline scripts/sfpshow:394]: @mihirpat1 : Please validate, add UT outcome of following CLIs:

specific port (optics EEPROM/DOM)
1) show int trans eeprom Ethernet<> -n asic<>
2) show int trans eeprom --dom Ethernet<> -n asic<>

all ports where optics inserted in LC/Fixed-chassis
3) show int trans eeprom
4)show int trans eeprom --dom

Comment (mihirpat1) [reply scripts/sfpshow:394]: I have now added the multi asic UT enclosure to https://github.com/sonic-net/sonic-platform-daemons/pull/320.
https://github.com/sonic-net/sonic-platform-daemons/files/10157449/Unit-test_400ZR_Multiasic.txt

**Review by shyam77git — COMMENTED** [review]
changeset and the multi-ASIC UT (I asked for) LGTM

**Review by shyam77git — APPROVED** [review]

**Review by prgeor — APPROVED** [review]

**Comment by mihirpat1** [comment]
> @mihirpat1 if no change can you specify that in these section: ` Previous command output (if the output of a command-line utility has changed)
> 
> New command output (if the output of a command-line utility has changed) `

I have updated the PR description now with the above info.

**Pull request merged**